src/bp/core/services/dialog/flow/service.ts

```diff
@@ -49,58 +54,56 @@ export class MutexError extends Error {
   type = MutexError.name
 }
 
-class FlowCache {
-  private _flows: Map<string, Map<string, FlowView>> = new Map()
+class ArrayCache<K, V> {
+  private array: V[] = []
 
```

Comment:
This `ArrayCache` should probably be in its own file, but I'm not sure where to put it